src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala

```diff
@@ -380,7 +378,12 @@ trait NamesDefaults { self: Analyzer =>
                   }
               })
               // cannot call blockTyper.typedBlock here, because the method expr might be partially applied only
-              val res = blockTyper.doTypedApply(tree, expr, refArgs, mode, pt)
+              val res =
+                blockTyper.doTypedApply(tree, expr, refArgs, mode, pt) match {
```

Comment:
Another way to skin the cat would be to avoid const-folding the `refArg` for the by-name param. The `refArg` has type `Seq[String("sl19")] @scala.reflect.internal.annotations.uncheckedBounds`, which folds.

Changing the code to `f(b = "sl19": String, a = 28)` turns it into `Seq[String] @scala.reflect.internal.annotations.uncheckedBounds`, which doesn't fold.

https://github.com/scala/scala/compare/2.13.x...lrytz:pr9662-alt?expand=1

Reply:
Speaking of by-name params, I realized that Scala 2 doesn't have by-name repeated params yet. I opened a ticket about that when I was learning me some scala.